app/src/main/java/mollie/tictactoe/activities/BoardActivity.java

```diff
@@ -0,0 +1,84 @@
+package mollie.tictactoe.activities;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+import mollie.tictactoe.GameHelper;
+import mollie.tictactoe.R;
+import mollie.tictactoe.StateManager;
+import mollie.tictactoe.ui.BoardView;
+import mollie.tictactoe.ui.CellButton;
+import mollie.tictactoe.ui.UIBoardManager;
+
+public class BoardActivity extends AppCompatActivity {
+
+    private GameHelper mGameHelper = new GameHelper();
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_board);
+        mGameHelper = new GameHelper(StateManager.restoreStateToGame(savedInstanceState));
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        UIBoardManager.updateUIBoard(mGameHelper.getBoard(), getView());
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        StateManager.getOutStateBundle(outState, mGameHelper.getGame());
+    }
+
+    public void placeMark(View view) {
+        CellButton button = (CellButton) view;
+        int position = getButtonPosition(button);
+        String mark = mGameHelper.playMove(position);
+        UIBoardManager.updateUI(mark, button);
+        if (mGameHelper.gameIsOver()) {
+            UIBoardManager.endGame(mGameHelper.getWinner(), getView(), getApplicationContext());
+            promptForPlayAgain().show();
+        }
+    }
+
+    private BoardView getView() {
+        View view = findViewById(R.id.board);
+        return ((BoardView) view);
+    }
+
+    private int getButtonPosition(CellButton button) {
+        return Integer.valueOf((String) button.getTag());
+    }
+
+    public Dialog promptForPlayAgain() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Would you like to play again?")
+                .setPositiveButton("yes", new PlayAgainClickListener())
+                .setNegativeButton("no", new CancelDialogListener());
```

Comment:
I would avoid any text in your code, using an .xml external file is better. It will allow you to things:
- avoid to re compile your project for any silly grammatical error
- internationalize your application latter if it's needed